dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -817,7 +817,11 @@ private Response getFileByToolVersionID(String registryId, String versionIdParam
      */
     private static String computeURLFromEntryAndRequestURI(String entry, String selfPath) {
         String url = ToolsImplCommon.getUrlFromId(config, entry);
-        return url + selfPath.split(URLEncoder.encode(entry, StandardCharsets.UTF_8))[1];
+        String[] splitPath = selfPath.split(URLEncoder.encode(entry, StandardCharsets.UTF_8));
```

Comment:
allergies messing  with me, but it took me too long to realize that this was string splitting based on the encoded id itself
i.e. this gets you a split array with the first part of the url in the 0th index and any query  parameters or other stuff that follows the ID  in the second part